balancer-js/src/modules/sor/pool-data/onChainData.ts

```diff
@@ -66,85 +66,134 @@ export async function getOnChainBalances<
-     *    balance to get the total "circulating supply". This is called the 'virtualSupply'.
-     *  - the Pool owes debt to the Protocol in the form of unminted BPT, which will be minted immediately before the
-     *    next join or exit. We need to take these into account since, even if they don't yet exist, they will
-     *    effectively be included in any Pool operation that involves BPT.
-     * In the vast majority of cases, this function should be used instead of `totalSupply()`.
-     */
-    if (pool.poolType === 'ComposableStable')
-      multiPool.call(
-        `${pool.id}.actualSupply`,
-        pool.address,
-        'getActualSupply'
-      );
-
-    // TO DO - Make this part of class to make more flexible?
-    if (
-      pool.poolType === 'Weighted' ||
-      pool.poolType === 'LiquidityBootstrapping' ||
-      pool.poolType === 'Investment'
-    ) {
-      multiPool.call(
-        `${pool.id}.weights`,
-        pool.address,
-        'getNormalizedWeights'
-      );
-      multiPool.call(
-        `${pool.id}.swapFee`,
-        pool.address,
-        'getSwapFeePercentage'
-      );
-    } else if (
-      pool.poolType === 'Stable' ||
-      pool.poolType === 'MetaStable' ||
-      pool.poolType === 'StablePhantom' ||
-      pool.poolType === 'ComposableStable'
-    ) {
-      // MetaStable & StablePhantom is the same as Stable for multicall purposes
-      multiPool.call(
-        `${pool.id}.amp`,
-        pool.address,
-        'getAmplificationParameter'
-      );
-      multiPool.call(
-        `${pool.id}.swapFee`,
-        pool.address,
-        'getSwapFeePercentage'
-      );
-    } else if (pool.poolType === 'Element') {
-      multiPool.call(`${pool.id}.swapFee`, pool.address, 'percentFee');
-    } else if (pool.poolType.toString().includes('Linear')) {
-      multiPool.call(
-        `${pool.id}.swapFee`,
-        pool.address,
-        'getSwapFeePercentage'
-      );
-
-      multiPool.call(`${pool.id}.targets`, pool.address, 'getTargets');
-      multiPool.call(`${pool.id}.rate`, pool.address, 'getWrappedTokenRate');
-    } else if (pool.poolType.toString().includes('Gyro')) {
-      multiPool.call(
-        `${pool.id}.swapFee`,
-        pool.address,
-        'getSwapFeePercentage'
-      );
-    }
-    if (pool.poolType.toString() === 'GyroE' && pool.poolTypeVersion === 2) {
-      multiPool.call(`${pool.id}.tokenRates`, pool.address, 'getTokenRates');
+    switch (pool.poolType) {
+      case 'LiquidityBootstrapping':
+      case 'Investment':
+      case 'Weighted':
+        multiPool.call(
+          `${pool.id}.swapFee`,
+          pool.address,
+          'getSwapFeePercentage'
+        );
+        multiPool.call(
+          `${pool.id}.weights`,
+          pool.address,
+          'getNormalizedWeights'
+        );
+        break;
+      case 'AaveLinear':
+      case 'BeefyLinear':
+      case 'EulerLinear':
+      case 'ERC4626Linear':
```

Comment:
I think its better to keep the check for `subgraphPools[index].poolType.includes('Linear')` here otherwise we have to manually update this file whenever a new Linear factory is added - unless there's a strong reason to make this change?